manifest: openthread upmerge up to commit `2e625bfe`

edmont opened the pull request:
Update manifest with OpenThread up to commit `2e625bfe` and make use of new `mIsHeaderUpdated` flag in TX frame metadata.

Signed-off-by: Eduardo Montoya <eduardo.montoya@nordicsemi.no>

hubertmis commented on subsys/net/lib/openthread/platform/radio.c:
```diff
@@ -303,6 +307,9 @@ void transmit_message(struct k_work *tx_job)
 {
+	sTransmitFrame.mInfo.mTxInfo.mIsSecurityProcessed =
+		net_pkt_ieee802154_frame_secured(tx_pkt);
+
```
I thought it's necessary to set `.mIsARetx` based on `_mac_hdr_rdy()` here as well. Is it omitted intentionally?

edmont replied:
It's intentional. OpenThread already infers the value of this flag from the transmission result:
- For direct transmissions, in `SubMac::HandleTransmitDone`
- For indirect transmissions, in `DataPollHandler::HandleFrameRequest`
- For CSL transmissions, in `CslTxScheduler::HandleFrameRequest`

If `.mIsARetx` is set based on `_mac_hdr_rdy()` even successful transmissions will be marked as retransmissions and wrong behavior will occur, like a CSL Receiver sending a Data Poll to re-synchronize the CSL IEs with the Parent.

hubertmis replied:
But it would work incorrectly. OT expects that frame with `.mIsARetx` flag set has correct frame counter and CSL IE values set. However, if `_mac_hdr_rdy()` is `false`, then these fields in the frame have incorrect value. It causes inconsistent state of frame data and metadata in OpenThread MAC.

I'm looking at OpenThread implementation for possible solutions

edmont replied:
Solution provided in https://github.com/openthread/openthread/pull/6849

edmont committed:
net: openthread: propagate security and header updated flags

When transmitting a frame, inform the radio driver whether
security processing and/or header updates are needed or not.

When a frame was transmitted, inform back to OpenThread whether
the security procedure and/or header updates were completed for
the frame or not.

Signed-off-by: Eduardo Montoya <eduardo.montoya@nordicsemi.no>

edmont committed:
net: openthread: add Netdata Publisher Kconfig option

Enable new feature from the upmerge:
- OPENTHREAD_NETDATA_PUBLISHER

Signed-off-by: Eduardo Montoya <eduardo.montoya@nordicsemi.no>

rlubos commented on west.yml:
```diff
@@ -183,7 +183,7 @@ manifest:
       path: modules/lib/open-amp
     - name: openthread
-      revision: 542b14a5bc5b38f29e2cab892c66da670a524b05
+      revision: pull/46/head
```
Please update SHA: `6e88d170bb7567d5e9638168e597f602ff3b6046`

edmont committed:
manifest: openthread upmerge up to commit `2e625bfe`

Update manifest with OpenThread up to commit `2e625bfe`.

Signed-off-by: Eduardo Montoya <eduardo.montoya@nordicsemi.no>
